Automattic/jetpack
# Blocks: introduce new pattern to demonstrate some of our blocks

Fixes #35539

## Proposed changes:

This commit introduces 3 things:

1. A new directory and loading mechanism allowing us to add more patterns in the future.
2. A new Pattern category, "Jetpack".
3. A first pattern of a group of 3 Jetpack blocks: a sharing buttons block, a Like Block, and a Related Posts block.

<img width="732" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/8bcd8137-332c-4781-ade3-dc8b53f96770">

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

> [!NOTE]
> This is better tested on a site where you have some published posts already, so related posts can appear.

* Go to Appearance > Themes and activate a block-based theme like Twenty Twenty Four.
* Go to Appearance > Site Editor > Templates > Single Posts
* Add a pattern below the post content
    * Select the new pattern from the new "Jetpack" category.
